File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/events/retired_framework.rb
module ComplianceDomain
  class RegulatoryFramework
    module Events
      class RetiredFramework
        attr_reader :aggregate_id, :framework_id, :name, :jurisdiction, :version, :effective_date, :authority, :requirements, :status, :occurred_at

        def initialize(aggregate_id: nil, framework_id: nil, name: nil, jurisdiction: nil, version: nil, effective_date: nil, authority: nil, requirements: nil, status: nil)
          @aggregate_id = aggregate_id
          @framework_id = framework_id
          @name = name
          @jurisdiction = jurisdiction
          @version = version
          @effective_date = effective_date
          @authority = authority
          @requirements = requirements
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end

File: examples/governance_static_go/domain/retired_framework.go
package domain

import "time"

type RetiredFramework struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	FrameworkId string `json:"framework_id"`
	Name string `json:"name"`
	Jurisdiction string `json:"jurisdiction"`
	Version string `json:"version"`
	EffectiveDate time.Time `json:"effective_date"`
	Authority string `json:"authority"`
	Requirements []FrameworkRequirement `json:"requirements"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RetiredFramework) EventName() string { return "RetiredFramework" }

func (e RetiredFramework) GetOccurredAt() time.Time { return e.OccurredAt }
